// @aws-sdk/client-elastic-beanstalk/dist-es/commands/DeletePlatformVersionCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeletePlatformVersionRequestFilterSensitiveLog, DeletePlatformVersionResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDeletePlatformVersionCommand, serializeAws_queryDeletePlatformVersionCommand, } from "../protocols/Aws_query";
export class DeletePlatformVersionCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, DeletePlatformVersionCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "ElasticBeanstalkClient";
        const commandName = "DeletePlatformVersionCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DeletePlatformVersionRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeletePlatformVersionResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryDeletePlatformVersionCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryDeletePlatformVersionCommand(output, context);
    }
}
