// @aws-sdk/client-elastic-beanstalk/dist-es/commands/UpdateEnvironmentCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { EnvironmentDescriptionFilterSensitiveLog, UpdateEnvironmentMessageFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryUpdateEnvironmentCommand, serializeAws_queryUpdateEnvironmentCommand, } from "../protocols/Aws_query";
export class UpdateEnvironmentCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, UpdateEnvironmentCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "ElasticBeanstalkClient";
        const commandName = "UpdateEnvironmentCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: UpdateEnvironmentMessageFilterSensitiveLog,
            outputFilterSensitiveLog: EnvironmentDescriptionFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryUpdateEnvironmentCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryUpdateEnvironmentCommand(output, context);
    }
}
