// @aws-sdk/client-elastic-beanstalk/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadBalancerDescriptionFilterSensitiveLog = exports.ListenerFilterSensitiveLog = exports.EnvironmentLinkFilterSensitiveLog = exports.ComposeEnvironmentsMessageFilterSensitiveLog = exports.CheckDNSAvailabilityResultMessageFilterSensitiveLog = exports.CheckDNSAvailabilityMessageFilterSensitiveLog = exports.SolutionStackDescriptionFilterSensitiveLog = exports.AutoScalingGroupFilterSensitiveLog = exports.AssociateEnvironmentOperationsRoleMessageFilterSensitiveLog = exports.ApplyEnvironmentManagedActionResultFilterSensitiveLog = exports.ApplyEnvironmentManagedActionRequestFilterSensitiveLog = exports.ApplicationVersionDescriptionsMessageFilterSensitiveLog = exports.ApplicationVersionDescriptionMessageFilterSensitiveLog = exports.ApplicationVersionDescriptionFilterSensitiveLog = exports.S3LocationFilterSensitiveLog = exports.SourceBuildInformationFilterSensitiveLog = exports.ApplicationResourceLifecycleDescriptionMessageFilterSensitiveLog = exports.ApplicationMetricsFilterSensitiveLog = exports.StatusCodesFilterSensitiveLog = exports.LatencyFilterSensitiveLog = exports.ApplicationDescriptionsMessageFilterSensitiveLog = exports.ApplicationDescriptionMessageFilterSensitiveLog = exports.ApplicationDescriptionFilterSensitiveLog = exports.ApplicationResourceLifecycleConfigFilterSensitiveLog = exports.ApplicationVersionLifecycleConfigFilterSensitiveLog = exports.MaxCountRuleFilterSensitiveLog = exports.MaxAgeRuleFilterSensitiveLog = exports.AbortEnvironmentUpdateMessageFilterSensitiveLog = exports.TooManyTagsException = exports.ResourceTypeNotSupportedException = exports.ResourceNotFoundException = exports.InstancesHealthAttribute = exports.InvalidRequestException = exports.EnvironmentHealthAttribute = exports.PlatformVersionStillReferencedException = exports.SourceBundleDeletionException = exports.OperationInProgressException = exports.S3SubscriptionRequiredException = exports.TooManyPlatformsException = exports.TooManyConfigurationTemplatesException = exports.TooManyBucketsException = exports.TooManyApplicationVersionsException = exports.S3LocationNotInServiceRegionException = exports.ComputeType = exports.CodeBuildNotInServiceRegionException = exports.TooManyApplicationsException = exports.TooManyEnvironmentsException = exports.ManagedActionInvalidStateException = exports.ElasticBeanstalkServiceException = exports.InsufficientPrivilegesException = void 0;
exports.LoadBalancerFilterSensitiveLog = exports.LaunchTemplateFilterSensitiveLog = exports.LaunchConfigurationFilterSensitiveLog = exports.InstanceFilterSensitiveLog = exports.DescribeEnvironmentResourcesMessageFilterSensitiveLog = exports.DescribeEnvironmentManagedActionsResultFilterSensitiveLog = exports.ManagedActionFilterSensitiveLog = exports.DescribeEnvironmentManagedActionsRequestFilterSensitiveLog = exports.DescribeEnvironmentManagedActionHistoryResultFilterSensitiveLog = exports.ManagedActionHistoryItemFilterSensitiveLog = exports.DescribeEnvironmentManagedActionHistoryRequestFilterSensitiveLog = exports.DescribeEnvironmentHealthResultFilterSensitiveLog = exports.InstanceHealthSummaryFilterSensitiveLog = exports.DescribeEnvironmentHealthRequestFilterSensitiveLog = exports.DescribeConfigurationSettingsMessageFilterSensitiveLog = exports.ConfigurationSettingsDescriptionsFilterSensitiveLog = exports.DescribeConfigurationOptionsMessageFilterSensitiveLog = exports.ConfigurationOptionsDescriptionFilterSensitiveLog = exports.ConfigurationOptionDescriptionFilterSensitiveLog = exports.OptionRestrictionRegexFilterSensitiveLog = exports.DescribeApplicationVersionsMessageFilterSensitiveLog = exports.DescribeApplicationsMessageFilterSensitiveLog = exports.DescribeAccountAttributesResultFilterSensitiveLog = exports.ResourceQuotasFilterSensitiveLog = exports.ResourceQuotaFilterSensitiveLog = exports.DeletePlatformVersionResultFilterSensitiveLog = exports.DeletePlatformVersionRequestFilterSensitiveLog = exports.DeleteEnvironmentConfigurationMessageFilterSensitiveLog = exports.DeleteConfigurationTemplateMessageFilterSensitiveLog = exports.DeleteApplicationVersionMessageFilterSensitiveLog = exports.DeleteApplicationMessageFilterSensitiveLog = exports.CreateStorageLocationResultMessageFilterSensitiveLog = exports.CreatePlatformVersionResultFilterSensitiveLog = exports.PlatformSummaryFilterSensitiveLog = exports.BuilderFilterSensitiveLog = exports.CreatePlatformVersionRequestFilterSensitiveLog = exports.CreateEnvironmentMessageFilterSensitiveLog = exports.OptionSpecificationFilterSensitiveLog = exports.CreateConfigurationTemplateMessageFilterSensitiveLog = exports.SourceConfigurationFilterSensitiveLog = exports.ConfigurationSettingsDescriptionFilterSensitiveLog = exports.ConfigurationOptionSettingFilterSensitiveLog = exports.CreateApplicationVersionMessageFilterSensitiveLog = exports.BuildConfigurationFilterSensitiveLog = exports.CreateApplicationMessageFilterSensitiveLog = exports.TagFilterSensitiveLog = exports.EnvironmentDescriptionsMessageFilterSensitiveLog = exports.EnvironmentDescriptionFilterSensitiveLog = exports.EnvironmentTierFilterSensitiveLog = exports.EnvironmentResourcesDescriptionFilterSensitiveLog = void 0;
exports.ValidateConfigurationSettingsMessageFilterSensitiveLog = exports.ConfigurationSettingsValidationMessagesFilterSensitiveLog = exports.ValidationMessageFilterSensitiveLog = exports.UpdateTagsForResourceMessageFilterSensitiveLog = exports.UpdateEnvironmentMessageFilterSensitiveLog = exports.UpdateConfigurationTemplateMessageFilterSensitiveLog = exports.UpdateApplicationVersionMessageFilterSensitiveLog = exports.UpdateApplicationResourceLifecycleMessageFilterSensitiveLog = exports.UpdateApplicationMessageFilterSensitiveLog = exports.TerminateEnvironmentMessageFilterSensitiveLog = exports.SwapEnvironmentCNAMEsMessageFilterSensitiveLog = exports.RetrieveEnvironmentInfoResultMessageFilterSensitiveLog = exports.EnvironmentInfoDescriptionFilterSensitiveLog = exports.RetrieveEnvironmentInfoMessageFilterSensitiveLog = exports.RestartAppServerMessageFilterSensitiveLog = exports.RequestEnvironmentInfoMessageFilterSensitiveLog = exports.RebuildEnvironmentMessageFilterSensitiveLog = exports.ResourceTagsDescriptionMessageFilterSensitiveLog = exports.ListTagsForResourceMessageFilterSensitiveLog = exports.ListPlatformVersionsResultFilterSensitiveLog = exports.ListPlatformVersionsRequestFilterSensitiveLog = exports.PlatformFilterFilterSensitiveLog = exports.ListPlatformBranchesResultFilterSensitiveLog = exports.PlatformBranchSummaryFilterSensitiveLog = exports.ListPlatformBranchesRequestFilterSensitiveLog = exports.SearchFilterFilterSensitiveLog = exports.ListAvailableSolutionStacksResultMessageFilterSensitiveLog = exports.DisassociateEnvironmentOperationsRoleMessageFilterSensitiveLog = exports.DescribePlatformVersionResultFilterSensitiveLog = exports.PlatformDescriptionFilterSensitiveLog = exports.PlatformProgrammingLanguageFilterSensitiveLog = exports.PlatformFrameworkFilterSensitiveLog = exports.CustomAmiFilterSensitiveLog = exports.DescribePlatformVersionRequestFilterSensitiveLog = exports.DescribeInstancesHealthResultFilterSensitiveLog = exports.SingleInstanceHealthFilterSensitiveLog = exports.SystemStatusFilterSensitiveLog = exports.CPUUtilizationFilterSensitiveLog = exports.DeploymentFilterSensitiveLog = exports.DescribeInstancesHealthRequestFilterSensitiveLog = exports.EventDescriptionsMessageFilterSensitiveLog = exports.EventDescriptionFilterSensitiveLog = exports.DescribeEventsMessageFilterSensitiveLog = exports.DescribeEnvironmentsMessageFilterSensitiveLog = exports.EnvironmentResourceDescriptionsMessageFilterSensitiveLog = exports.EnvironmentResourceDescriptionFilterSensitiveLog = exports.TriggerFilterSensitiveLog = exports.QueueFilterSensitiveLog = void 0;
const ElasticBeanstalkServiceException_1 = require("./ElasticBeanstalkServiceException");
class InsufficientPrivilegesException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "InsufficientPrivilegesException",
            $fault: "client",
            ...opts,
        });
        this.name = "InsufficientPrivilegesException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InsufficientPrivilegesException.prototype);
    }
}
exports.InsufficientPrivilegesException = InsufficientPrivilegesException;
class ElasticBeanstalkServiceException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "ElasticBeanstalkServiceException",
            $fault: "client",
            ...opts,
        });
        this.name = "ElasticBeanstalkServiceException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ElasticBeanstalkServiceException.prototype);
    }
}
exports.ElasticBeanstalkServiceException = ElasticBeanstalkServiceException;
class ManagedActionInvalidStateException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "ManagedActionInvalidStateException",
            $fault: "client",
            ...opts,
        });
        this.name = "ManagedActionInvalidStateException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ManagedActionInvalidStateException.prototype);
    }
}
exports.ManagedActionInvalidStateException = ManagedActionInvalidStateException;
class TooManyEnvironmentsException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "TooManyEnvironmentsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyEnvironmentsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyEnvironmentsException.prototype);
    }
}
exports.TooManyEnvironmentsException = TooManyEnvironmentsException;
class TooManyApplicationsException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "TooManyApplicationsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyApplicationsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyApplicationsException.prototype);
    }
}
exports.TooManyApplicationsException = TooManyApplicationsException;
class CodeBuildNotInServiceRegionException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "CodeBuildNotInServiceRegionException",
            $fault: "client",
            ...opts,
        });
        this.name = "CodeBuildNotInServiceRegionException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CodeBuildNotInServiceRegionException.prototype);
    }
}
exports.CodeBuildNotInServiceRegionException = CodeBuildNotInServiceRegionException;
var ComputeType;
(function (ComputeType) {
    ComputeType["BUILD_GENERAL1_LARGE"] = "BUILD_GENERAL1_LARGE";
    ComputeType["BUILD_GENERAL1_MEDIUM"] = "BUILD_GENERAL1_MEDIUM";
    ComputeType["BUILD_GENERAL1_SMALL"] = "BUILD_GENERAL1_SMALL";
})(ComputeType = exports.ComputeType || (exports.ComputeType = {}));
class S3LocationNotInServiceRegionException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "S3LocationNotInServiceRegionException",
            $fault: "client",
            ...opts,
        });
        this.name = "S3LocationNotInServiceRegionException";
        this.$fault = "client";
        Object.setPrototypeOf(this, S3LocationNotInServiceRegionException.prototype);
    }
}
exports.S3LocationNotInServiceRegionException = S3LocationNotInServiceRegionException;
class TooManyApplicationVersionsException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "TooManyApplicationVersionsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyApplicationVersionsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyApplicationVersionsException.prototype);
    }
}
exports.TooManyApplicationVersionsException = TooManyApplicationVersionsException;
class TooManyBucketsException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "TooManyBucketsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyBucketsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyBucketsException.prototype);
    }
}
exports.TooManyBucketsException = TooManyBucketsException;
class TooManyConfigurationTemplatesException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "TooManyConfigurationTemplatesException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyConfigurationTemplatesException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyConfigurationTemplatesException.prototype);
    }
}
exports.TooManyConfigurationTemplatesException = TooManyConfigurationTemplatesException;
class TooManyPlatformsException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "TooManyPlatformsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyPlatformsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyPlatformsException.prototype);
    }
}
exports.TooManyPlatformsException = TooManyPlatformsException;
class S3SubscriptionRequiredException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "S3SubscriptionRequiredException",
            $fault: "client",
            ...opts,
        });
        this.name = "S3SubscriptionRequiredException";
        this.$fault = "client";
        Object.setPrototypeOf(this, S3SubscriptionRequiredException.prototype);
    }
}
exports.S3SubscriptionRequiredException = S3SubscriptionRequiredException;
class OperationInProgressException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "OperationInProgressException",
            $fault: "client",
            ...opts,
        });
        this.name = "OperationInProgressException";
        this.$fault = "client";
        Object.setPrototypeOf(this, OperationInProgressException.prototype);
    }
}
exports.OperationInProgressException = OperationInProgressException;
class SourceBundleDeletionException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "SourceBundleDeletionException",
            $fault: "client",
            ...opts,
        });
        this.name = "SourceBundleDeletionException";
        this.$fault = "client";
        Object.setPrototypeOf(this, SourceBundleDeletionException.prototype);
    }
}
exports.SourceBundleDeletionException = SourceBundleDeletionException;
class PlatformVersionStillReferencedException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "PlatformVersionStillReferencedException",
            $fault: "client",
            ...opts,
        });
        this.name = "PlatformVersionStillReferencedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, PlatformVersionStillReferencedException.prototype);
    }
}
exports.PlatformVersionStillReferencedException = PlatformVersionStillReferencedException;
var EnvironmentHealthAttribute;
(function (EnvironmentHealthAttribute) {
    EnvironmentHealthAttribute["All"] = "All";
    EnvironmentHealthAttribute["ApplicationMetrics"] = "ApplicationMetrics";
    EnvironmentHealthAttribute["Causes"] = "Causes";
    EnvironmentHealthAttribute["Color"] = "Color";
    EnvironmentHealthAttribute["HealthStatus"] = "HealthStatus";
    EnvironmentHealthAttribute["InstancesHealth"] = "InstancesHealth";
    EnvironmentHealthAttribute["RefreshedAt"] = "RefreshedAt";
    EnvironmentHealthAttribute["Status"] = "Status";
})(EnvironmentHealthAttribute = exports.EnvironmentHealthAttribute || (exports.EnvironmentHealthAttribute = {}));
class InvalidRequestException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "InvalidRequestException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidRequestException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidRequestException.prototype);
    }
}
exports.InvalidRequestException = InvalidRequestException;
var InstancesHealthAttribute;
(function (InstancesHealthAttribute) {
    InstancesHealthAttribute["All"] = "All";
    InstancesHealthAttribute["ApplicationMetrics"] = "ApplicationMetrics";
    InstancesHealthAttribute["AvailabilityZone"] = "AvailabilityZone";
    InstancesHealthAttribute["Causes"] = "Causes";
    InstancesHealthAttribute["Color"] = "Color";
    InstancesHealthAttribute["Deployment"] = "Deployment";
    InstancesHealthAttribute["HealthStatus"] = "HealthStatus";
    InstancesHealthAttribute["InstanceType"] = "InstanceType";
    InstancesHealthAttribute["LaunchedAt"] = "LaunchedAt";
    InstancesHealthAttribute["RefreshedAt"] = "RefreshedAt";
    InstancesHealthAttribute["System"] = "System";
})(InstancesHealthAttribute = exports.InstancesHealthAttribute || (exports.InstancesHealthAttribute = {}));
class ResourceNotFoundException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class ResourceTypeNotSupportedException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "ResourceTypeNotSupportedException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceTypeNotSupportedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceTypeNotSupportedException.prototype);
    }
}
exports.ResourceTypeNotSupportedException = ResourceTypeNotSupportedException;
class TooManyTagsException extends ElasticBeanstalkServiceException_1.ElasticBeanstalkServiceException {
    constructor(opts) {
        super({
            name: "TooManyTagsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyTagsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyTagsException.prototype);
    }
}
exports.TooManyTagsException = TooManyTagsException;
const AbortEnvironmentUpdateMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AbortEnvironmentUpdateMessageFilterSensitiveLog = AbortEnvironmentUpdateMessageFilterSensitiveLog;
const MaxAgeRuleFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MaxAgeRuleFilterSensitiveLog = MaxAgeRuleFilterSensitiveLog;
const MaxCountRuleFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MaxCountRuleFilterSensitiveLog = MaxCountRuleFilterSensitiveLog;
const ApplicationVersionLifecycleConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationVersionLifecycleConfigFilterSensitiveLog = ApplicationVersionLifecycleConfigFilterSensitiveLog;
const ApplicationResourceLifecycleConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationResourceLifecycleConfigFilterSensitiveLog = ApplicationResourceLifecycleConfigFilterSensitiveLog;
const ApplicationDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationDescriptionFilterSensitiveLog = ApplicationDescriptionFilterSensitiveLog;
const ApplicationDescriptionMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationDescriptionMessageFilterSensitiveLog = ApplicationDescriptionMessageFilterSensitiveLog;
const ApplicationDescriptionsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationDescriptionsMessageFilterSensitiveLog = ApplicationDescriptionsMessageFilterSensitiveLog;
const LatencyFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LatencyFilterSensitiveLog = LatencyFilterSensitiveLog;
const StatusCodesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StatusCodesFilterSensitiveLog = StatusCodesFilterSensitiveLog;
const ApplicationMetricsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationMetricsFilterSensitiveLog = ApplicationMetricsFilterSensitiveLog;
const ApplicationResourceLifecycleDescriptionMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationResourceLifecycleDescriptionMessageFilterSensitiveLog = ApplicationResourceLifecycleDescriptionMessageFilterSensitiveLog;
const SourceBuildInformationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SourceBuildInformationFilterSensitiveLog = SourceBuildInformationFilterSensitiveLog;
const S3LocationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.S3LocationFilterSensitiveLog = S3LocationFilterSensitiveLog;
const ApplicationVersionDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationVersionDescriptionFilterSensitiveLog = ApplicationVersionDescriptionFilterSensitiveLog;
const ApplicationVersionDescriptionMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationVersionDescriptionMessageFilterSensitiveLog = ApplicationVersionDescriptionMessageFilterSensitiveLog;
const ApplicationVersionDescriptionsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplicationVersionDescriptionsMessageFilterSensitiveLog = ApplicationVersionDescriptionsMessageFilterSensitiveLog;
const ApplyEnvironmentManagedActionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplyEnvironmentManagedActionRequestFilterSensitiveLog = ApplyEnvironmentManagedActionRequestFilterSensitiveLog;
const ApplyEnvironmentManagedActionResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApplyEnvironmentManagedActionResultFilterSensitiveLog = ApplyEnvironmentManagedActionResultFilterSensitiveLog;
const AssociateEnvironmentOperationsRoleMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AssociateEnvironmentOperationsRoleMessageFilterSensitiveLog = AssociateEnvironmentOperationsRoleMessageFilterSensitiveLog;
const AutoScalingGroupFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AutoScalingGroupFilterSensitiveLog = AutoScalingGroupFilterSensitiveLog;
const SolutionStackDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SolutionStackDescriptionFilterSensitiveLog = SolutionStackDescriptionFilterSensitiveLog;
const CheckDNSAvailabilityMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CheckDNSAvailabilityMessageFilterSensitiveLog = CheckDNSAvailabilityMessageFilterSensitiveLog;
const CheckDNSAvailabilityResultMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CheckDNSAvailabilityResultMessageFilterSensitiveLog = CheckDNSAvailabilityResultMessageFilterSensitiveLog;
const ComposeEnvironmentsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ComposeEnvironmentsMessageFilterSensitiveLog = ComposeEnvironmentsMessageFilterSensitiveLog;
const EnvironmentLinkFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EnvironmentLinkFilterSensitiveLog = EnvironmentLinkFilterSensitiveLog;
const ListenerFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListenerFilterSensitiveLog = ListenerFilterSensitiveLog;
const LoadBalancerDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LoadBalancerDescriptionFilterSensitiveLog = LoadBalancerDescriptionFilterSensitiveLog;
const EnvironmentResourcesDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EnvironmentResourcesDescriptionFilterSensitiveLog = EnvironmentResourcesDescriptionFilterSensitiveLog;
const EnvironmentTierFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EnvironmentTierFilterSensitiveLog = EnvironmentTierFilterSensitiveLog;
const EnvironmentDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EnvironmentDescriptionFilterSensitiveLog = EnvironmentDescriptionFilterSensitiveLog;
const EnvironmentDescriptionsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EnvironmentDescriptionsMessageFilterSensitiveLog = EnvironmentDescriptionsMessageFilterSensitiveLog;
const TagFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagFilterSensitiveLog = TagFilterSensitiveLog;
const CreateApplicationMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateApplicationMessageFilterSensitiveLog = CreateApplicationMessageFilterSensitiveLog;
const BuildConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BuildConfigurationFilterSensitiveLog = BuildConfigurationFilterSensitiveLog;
const CreateApplicationVersionMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateApplicationVersionMessageFilterSensitiveLog = CreateApplicationVersionMessageFilterSensitiveLog;
const ConfigurationOptionSettingFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConfigurationOptionSettingFilterSensitiveLog = ConfigurationOptionSettingFilterSensitiveLog;
const ConfigurationSettingsDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConfigurationSettingsDescriptionFilterSensitiveLog = ConfigurationSettingsDescriptionFilterSensitiveLog;
const SourceConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SourceConfigurationFilterSensitiveLog = SourceConfigurationFilterSensitiveLog;
const CreateConfigurationTemplateMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateConfigurationTemplateMessageFilterSensitiveLog = CreateConfigurationTemplateMessageFilterSensitiveLog;
const OptionSpecificationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.OptionSpecificationFilterSensitiveLog = OptionSpecificationFilterSensitiveLog;
const CreateEnvironmentMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateEnvironmentMessageFilterSensitiveLog = CreateEnvironmentMessageFilterSensitiveLog;
const CreatePlatformVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreatePlatformVersionRequestFilterSensitiveLog = CreatePlatformVersionRequestFilterSensitiveLog;
const BuilderFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BuilderFilterSensitiveLog = BuilderFilterSensitiveLog;
const PlatformSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PlatformSummaryFilterSensitiveLog = PlatformSummaryFilterSensitiveLog;
const CreatePlatformVersionResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreatePlatformVersionResultFilterSensitiveLog = CreatePlatformVersionResultFilterSensitiveLog;
const CreateStorageLocationResultMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateStorageLocationResultMessageFilterSensitiveLog = CreateStorageLocationResultMessageFilterSensitiveLog;
const DeleteApplicationMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteApplicationMessageFilterSensitiveLog = DeleteApplicationMessageFilterSensitiveLog;
const DeleteApplicationVersionMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteApplicationVersionMessageFilterSensitiveLog = DeleteApplicationVersionMessageFilterSensitiveLog;
const DeleteConfigurationTemplateMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteConfigurationTemplateMessageFilterSensitiveLog = DeleteConfigurationTemplateMessageFilterSensitiveLog;
const DeleteEnvironmentConfigurationMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteEnvironmentConfigurationMessageFilterSensitiveLog = DeleteEnvironmentConfigurationMessageFilterSensitiveLog;
const DeletePlatformVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeletePlatformVersionRequestFilterSensitiveLog = DeletePlatformVersionRequestFilterSensitiveLog;
const DeletePlatformVersionResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeletePlatformVersionResultFilterSensitiveLog = DeletePlatformVersionResultFilterSensitiveLog;
const ResourceQuotaFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResourceQuotaFilterSensitiveLog = ResourceQuotaFilterSensitiveLog;
const ResourceQuotasFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResourceQuotasFilterSensitiveLog = ResourceQuotasFilterSensitiveLog;
const DescribeAccountAttributesResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeAccountAttributesResultFilterSensitiveLog = DescribeAccountAttributesResultFilterSensitiveLog;
const DescribeApplicationsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeApplicationsMessageFilterSensitiveLog = DescribeApplicationsMessageFilterSensitiveLog;
const DescribeApplicationVersionsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeApplicationVersionsMessageFilterSensitiveLog = DescribeApplicationVersionsMessageFilterSensitiveLog;
const OptionRestrictionRegexFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.OptionRestrictionRegexFilterSensitiveLog = OptionRestrictionRegexFilterSensitiveLog;
const ConfigurationOptionDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConfigurationOptionDescriptionFilterSensitiveLog = ConfigurationOptionDescriptionFilterSensitiveLog;
const ConfigurationOptionsDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConfigurationOptionsDescriptionFilterSensitiveLog = ConfigurationOptionsDescriptionFilterSensitiveLog;
const DescribeConfigurationOptionsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeConfigurationOptionsMessageFilterSensitiveLog = DescribeConfigurationOptionsMessageFilterSensitiveLog;
const ConfigurationSettingsDescriptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConfigurationSettingsDescriptionsFilterSensitiveLog = ConfigurationSettingsDescriptionsFilterSensitiveLog;
const DescribeConfigurationSettingsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeConfigurationSettingsMessageFilterSensitiveLog = DescribeConfigurationSettingsMessageFilterSensitiveLog;
const DescribeEnvironmentHealthRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeEnvironmentHealthRequestFilterSensitiveLog = DescribeEnvironmentHealthRequestFilterSensitiveLog;
const InstanceHealthSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.InstanceHealthSummaryFilterSensitiveLog = InstanceHealthSummaryFilterSensitiveLog;
const DescribeEnvironmentHealthResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeEnvironmentHealthResultFilterSensitiveLog = DescribeEnvironmentHealthResultFilterSensitiveLog;
const DescribeEnvironmentManagedActionHistoryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeEnvironmentManagedActionHistoryRequestFilterSensitiveLog = DescribeEnvironmentManagedActionHistoryRequestFilterSensitiveLog;
const ManagedActionHistoryItemFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ManagedActionHistoryItemFilterSensitiveLog = ManagedActionHistoryItemFilterSensitiveLog;
const DescribeEnvironmentManagedActionHistoryResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeEnvironmentManagedActionHistoryResultFilterSensitiveLog = DescribeEnvironmentManagedActionHistoryResultFilterSensitiveLog;
const DescribeEnvironmentManagedActionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeEnvironmentManagedActionsRequestFilterSensitiveLog = DescribeEnvironmentManagedActionsRequestFilterSensitiveLog;
const ManagedActionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ManagedActionFilterSensitiveLog = ManagedActionFilterSensitiveLog;
const DescribeEnvironmentManagedActionsResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeEnvironmentManagedActionsResultFilterSensitiveLog = DescribeEnvironmentManagedActionsResultFilterSensitiveLog;
const DescribeEnvironmentResourcesMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeEnvironmentResourcesMessageFilterSensitiveLog = DescribeEnvironmentResourcesMessageFilterSensitiveLog;
const InstanceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.InstanceFilterSensitiveLog = InstanceFilterSensitiveLog;
const LaunchConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LaunchConfigurationFilterSensitiveLog = LaunchConfigurationFilterSensitiveLog;
const LaunchTemplateFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LaunchTemplateFilterSensitiveLog = LaunchTemplateFilterSensitiveLog;
const LoadBalancerFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LoadBalancerFilterSensitiveLog = LoadBalancerFilterSensitiveLog;
const QueueFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueueFilterSensitiveLog = QueueFilterSensitiveLog;
const TriggerFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TriggerFilterSensitiveLog = TriggerFilterSensitiveLog;
const EnvironmentResourceDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EnvironmentResourceDescriptionFilterSensitiveLog = EnvironmentResourceDescriptionFilterSensitiveLog;
const EnvironmentResourceDescriptionsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EnvironmentResourceDescriptionsMessageFilterSensitiveLog = EnvironmentResourceDescriptionsMessageFilterSensitiveLog;
const DescribeEnvironmentsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeEnvironmentsMessageFilterSensitiveLog = DescribeEnvironmentsMessageFilterSensitiveLog;
const DescribeEventsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeEventsMessageFilterSensitiveLog = DescribeEventsMessageFilterSensitiveLog;
const EventDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EventDescriptionFilterSensitiveLog = EventDescriptionFilterSensitiveLog;
const EventDescriptionsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EventDescriptionsMessageFilterSensitiveLog = EventDescriptionsMessageFilterSensitiveLog;
const DescribeInstancesHealthRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeInstancesHealthRequestFilterSensitiveLog = DescribeInstancesHealthRequestFilterSensitiveLog;
const DeploymentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeploymentFilterSensitiveLog = DeploymentFilterSensitiveLog;
const CPUUtilizationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CPUUtilizationFilterSensitiveLog = CPUUtilizationFilterSensitiveLog;
const SystemStatusFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SystemStatusFilterSensitiveLog = SystemStatusFilterSensitiveLog;
const SingleInstanceHealthFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SingleInstanceHealthFilterSensitiveLog = SingleInstanceHealthFilterSensitiveLog;
const DescribeInstancesHealthResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribeInstancesHealthResultFilterSensitiveLog = DescribeInstancesHealthResultFilterSensitiveLog;
const DescribePlatformVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribePlatformVersionRequestFilterSensitiveLog = DescribePlatformVersionRequestFilterSensitiveLog;
const CustomAmiFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CustomAmiFilterSensitiveLog = CustomAmiFilterSensitiveLog;
const PlatformFrameworkFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PlatformFrameworkFilterSensitiveLog = PlatformFrameworkFilterSensitiveLog;
const PlatformProgrammingLanguageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PlatformProgrammingLanguageFilterSensitiveLog = PlatformProgrammingLanguageFilterSensitiveLog;
const PlatformDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PlatformDescriptionFilterSensitiveLog = PlatformDescriptionFilterSensitiveLog;
const DescribePlatformVersionResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DescribePlatformVersionResultFilterSensitiveLog = DescribePlatformVersionResultFilterSensitiveLog;
const DisassociateEnvironmentOperationsRoleMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DisassociateEnvironmentOperationsRoleMessageFilterSensitiveLog = DisassociateEnvironmentOperationsRoleMessageFilterSensitiveLog;
const ListAvailableSolutionStacksResultMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListAvailableSolutionStacksResultMessageFilterSensitiveLog = ListAvailableSolutionStacksResultMessageFilterSensitiveLog;
const SearchFilterFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SearchFilterFilterSensitiveLog = SearchFilterFilterSensitiveLog;
const ListPlatformBranchesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListPlatformBranchesRequestFilterSensitiveLog = ListPlatformBranchesRequestFilterSensitiveLog;
const PlatformBranchSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PlatformBranchSummaryFilterSensitiveLog = PlatformBranchSummaryFilterSensitiveLog;
const ListPlatformBranchesResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListPlatformBranchesResultFilterSensitiveLog = ListPlatformBranchesResultFilterSensitiveLog;
const PlatformFilterFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PlatformFilterFilterSensitiveLog = PlatformFilterFilterSensitiveLog;
const ListPlatformVersionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListPlatformVersionsRequestFilterSensitiveLog = ListPlatformVersionsRequestFilterSensitiveLog;
const ListPlatformVersionsResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListPlatformVersionsResultFilterSensitiveLog = ListPlatformVersionsResultFilterSensitiveLog;
const ListTagsForResourceMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListTagsForResourceMessageFilterSensitiveLog = ListTagsForResourceMessageFilterSensitiveLog;
const ResourceTagsDescriptionMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResourceTagsDescriptionMessageFilterSensitiveLog = ResourceTagsDescriptionMessageFilterSensitiveLog;
const RebuildEnvironmentMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RebuildEnvironmentMessageFilterSensitiveLog = RebuildEnvironmentMessageFilterSensitiveLog;
const RequestEnvironmentInfoMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RequestEnvironmentInfoMessageFilterSensitiveLog = RequestEnvironmentInfoMessageFilterSensitiveLog;
const RestartAppServerMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RestartAppServerMessageFilterSensitiveLog = RestartAppServerMessageFilterSensitiveLog;
const RetrieveEnvironmentInfoMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RetrieveEnvironmentInfoMessageFilterSensitiveLog = RetrieveEnvironmentInfoMessageFilterSensitiveLog;
const EnvironmentInfoDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EnvironmentInfoDescriptionFilterSensitiveLog = EnvironmentInfoDescriptionFilterSensitiveLog;
const RetrieveEnvironmentInfoResultMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RetrieveEnvironmentInfoResultMessageFilterSensitiveLog = RetrieveEnvironmentInfoResultMessageFilterSensitiveLog;
const SwapEnvironmentCNAMEsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SwapEnvironmentCNAMEsMessageFilterSensitiveLog = SwapEnvironmentCNAMEsMessageFilterSensitiveLog;
const TerminateEnvironmentMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TerminateEnvironmentMessageFilterSensitiveLog = TerminateEnvironmentMessageFilterSensitiveLog;
const UpdateApplicationMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateApplicationMessageFilterSensitiveLog = UpdateApplicationMessageFilterSensitiveLog;
const UpdateApplicationResourceLifecycleMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateApplicationResourceLifecycleMessageFilterSensitiveLog = UpdateApplicationResourceLifecycleMessageFilterSensitiveLog;
const UpdateApplicationVersionMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateApplicationVersionMessageFilterSensitiveLog = UpdateApplicationVersionMessageFilterSensitiveLog;
const UpdateConfigurationTemplateMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateConfigurationTemplateMessageFilterSensitiveLog = UpdateConfigurationTemplateMessageFilterSensitiveLog;
const UpdateEnvironmentMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateEnvironmentMessageFilterSensitiveLog = UpdateEnvironmentMessageFilterSensitiveLog;
const UpdateTagsForResourceMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateTagsForResourceMessageFilterSensitiveLog = UpdateTagsForResourceMessageFilterSensitiveLog;
const ValidationMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ValidationMessageFilterSensitiveLog = ValidationMessageFilterSensitiveLog;
const ConfigurationSettingsValidationMessagesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConfigurationSettingsValidationMessagesFilterSensitiveLog = ConfigurationSettingsValidationMessagesFilterSensitiveLog;
const ValidateConfigurationSettingsMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ValidateConfigurationSettingsMessageFilterSensitiveLog = ValidateConfigurationSettingsMessageFilterSensitiveLog;
